File: 0392-is-subsequence/0392-is-subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        if(s.size() == 0 && t.size()==0){return true;}
        else if(s.size()>0 && t.size() == 0){return false;}
        else if(s.size() == 0 && t.size()>0){return true;}
        if(s.size() > t.size()){
            return false;
        }
        else if(s.size()==t.size()){
            
            for(int i =0;i<s.size();i++){
                // cout << s[i] <<" " <<t[i]<<endl;
                if(s[i]!=t[i]){return false;}
            }
            return true;
        }
        else{
                int i =0;
                for(int j = 0; j<t.size();j++){
                    if(t[j] == s[i]){
                        i = i+1;
                    }
                    if(i == s.size()){ return true;}
                }
        }
        return false;
    }
};

File: 0392-is-subsequence/0392-is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        
        sl = list(s)
        tl = list(t)
        j = 0
        for i, c in enumerate(s):
            # print(i, j, c, tl[j])
            while j < len(tl):
                if c == tl[j]:
                    # print(i, j, c, tl[j])
                    sl.pop()
                    j += 1
                    break
                j+=1
            # print(sl)
            # print(tl)
        print(sl)

        return True if sl == [] else False
            